### PR TITLE
Makefile: Add testing targets

### PR DESCRIPTION
**Pull request by nirarg**
**What this PR does / why we need it**:

Add three new targets to the Makefile: unit-tests, workflow-integration-tests and notification-integration-tests to make it easier to developer to run only tests
Also default exclude any test which has "integration" in its path, to avoid running integration tests by default, when running any phase from the parent module

Notice that notification-service integration tests are not merged yet, its going to be added in PR #274

**Change type**
- [ ] New feature
- [ ] Bug fix
- [ ] Unit tests
- [ ] Integration tests
- [ ] CI
- [ ] Documentation
- [ ] Auto generated SDK code

**Impacted services**
- [ ] Workflow Service
- [ ] Notifivcation Service

**Checklist**
- [x] Subject and description added to both, commit and PR.
- [ ] Relevant issues have been referenced.

### CODE REVIEW

**Comment by nirarg** [comment]
@gciavarrini FYI
Do you think there is any issue to have this merged with #281?

File: Makefile [hunk 1]
```diff
@@ -91,6 +91,15 @@ release: clean ## release and push modules to maven central
...
 fast-build: all
 
+unit-tests: mvn-checks java-checks
+	$(MAVEN) $(ARGS) surefire:test
+
+workflow-integration-tests: mvn-checks java-checks
+	$(MAVEN) $(ARGS) surefire:test '-Dtest=%regex[.*.examples.integration.*]' -pl workflow-examples
+
+notification-integration-tests: mvn-checks java-checks
+	$(MAVEN) $(ARGS) surefire:test '-Dtest=%regex[.*.notification.integration.*]' -pl workflow-examples
```
Comment (eloycoto) [inline Makefile:101]: I think that @gciavarrini changed the integration test to a custom folder, so there is no need at all, no?

Comment (nirarg) [reply Makefile:101]: You right
I will close this PR for now and will wait for Gloria's changes to be completed

**Comment by nirarg** [comment]
Closing this PR as all integration tests location is going to be changed in #281

**Pull request closed without merging**